### ISSUE TITLE
Turn TwitchAuthentication into a dependency of HomeViewModel

### ISSUE THREAD

**Issue by thePlebDev** [thePlebDev/Clicker#394]
# Proposed change
- So basically in the HomeViewModel I have this code:
```
 private val twitchAuthentication= TwitchAuthentication(
        twitchRepoImpl = twitchRepoImpl,
        tokenDataStore = tokenDataStore,
        scope = viewModelScope,
        _uiState = _uiState,
        _loginUIState = _loginUIState
    )
```
- Which is hard coded(not ideal). I want to rework this code into a dependency of HomeViewModel

# Why is this important
- This will make the code easier to test. However, some major refacting needs to be done first. I really do not like that we have to pass it the state and the scope. I am reading up on code design to see if I can fix that

# Additional context

**Comment by thePlebDev** [comment]
- So I don't think that is the right way to go about things.
- We turn TwitchAuthentication into its own viewmodel and make a composable similiar to this:
```
@Composable
fun SignUpScreen(viewModel: SignupViewModel = viewModel(factory = LocalViewModelFactory.current)) {
    val state by viewModel.state.observeAsState(SignUpState())

    if (state.isCredentialView) {
        CredentialEntryView(viewModel, state)
    } else {
        NameEntryView(viewModel, state)
    }
}
```
- The source to that ViewModel can be found [HERE](https://github.com/Greenstand/treetracker-android/blob/a87bc564603f721a2b717d9bcdc820e526f8a3a3/app/src/main/java/org/greenstand/android/TreeTracker/signup/SignUpScreen.kt#L29).
- that conditional (state.isCredentialView)  would be a state inside of the TwitchAuthenticationViewModel.
- Which we could change it to ,`if state.isvalidated`, Then we could show the other Composables and pass them the validated tokens from the TwitchAuthenticationViewModel

**Comment by thePlebDev** [comment]
closed with #396
